ckcp/openshift_dev_setup.sh

```diff
@@ -276,6 +276,12 @@ patches:
         description: This value refers to the hostAddress defined in the Route.
         value: $ckcp_route " >>"$ckcp_temp_dir/kustomization.yaml"
 
+	# Enable the "runtime/default" seccomp profile (required for restricted security context)
+	# This is enabled by default in restricted-v2 available in openshift >= 4.11
+	if ! (set +o pipefail && kubectl describe scc restricted restricted-v2 2>/dev/null |
+		grep -sq 'runtime/default') ; then
+		 kubectl patch scc restricted --type=merge -p '{"seccompProfiles":["runtime/default"]}'
```

Comment:
We should not patch SCCs installed by OpenShift. These changes will be removed by the cluster operators.

Reply:
So I would rather create a clone of that scc, patch it and assign it to the corresponding sa.
Correct ?

Reply:
Am I correct in that the reason this is here is because the restricted SCC doesn't allow any value to be set for the seccomp profile? 

Since this is only for `ckcp`, I think it makes sense to declare OpenShift 4.11 as a required prerequisite. We don't deploy `ckcp` in the actual service.

Reply:
Right @adambkaplan the restricted SCC in 4.10 does not allow have this [securityContext](https://github.com/openshift-pipelines/pipeline-service/pull/241/files#diff-6c3c2089e306854fc8e3fb3696029ed7f67f8424c5c0ad53836e5eeb27621cbeR79).
You may ask, why I am changing that in 4.10 while the target of the PR is 4.11? reason is that 4.11 pushes for a restricted security context(via warnings). 
The change works out of the box in 4.11 but the same securitContext requires patching scc in 4.10.
For the sake of consistency I was trying to patch scc in 4.10 to work around that.
We have multiple options:
- backport the security context change in 4.10. In this case I will need assistance from openshift support. I have tried this [doc](https://docs.openshift.com/container-platform/4.10/security/seccomp-profiles.html) and I did not succeed.
- Don't add securityContext in 4.10 and warn users that the securitycontext of kcp pod is not the same between 4.10 and 4.11
- Don't add security Context in 4.10 and declare that 4.11 is a required Prerequisite.

Reply:
@adambkaplan Correct, we would either upgrade the existing cluster, or request a new cluster to deploy the service on, before deprecating the 4.10 cluster.

Reply:
@Mo3m3n I'm fine with making 4.11 a prerequisite, we don't really have the bandwidth to support multiple OCP versions at the moment. By the time we might do it, 4.12 will be out and since we will probably limit support to `n` and `n-1` version, that would again rule out 4.10.